activerecord/lib/active_record/explain_subscriber.rb

```diff
@@ -29,6 +29,10 @@ def ignore_payload?(payload)
         !payload[:sql].match?(EXPLAINED_SQLS)
     end
 
+    def silenced?(event)
+      !ExplainRegistry.collect?
+    end
```

Comment:
Better than silencing it, we should just only subscribe on the event when we're collecting.

This means we'd flush the AS::N cache, but I think it's OK as you'd only realistically do this in development.